app/resources/admin.html.template

```diff
@@ -83,23 +94,42 @@
         return false;
       }
       used_languages.push(language_code);
-      languages_to_titles[language_code] = language_title;
     }
 
     // Copy the dropdowns and the titles into the text boxes that will
     // actually be POSTed (language_menu_options and repo_titles).
     $('language_menu_options').value = JSON.stringify(used_languages);
-    $('repo_titles').value = JSON.stringify(languages_to_titles);
-
+    var languages_to_values = {};
+    for(var id_num = 0; id_num < CUSTOM_MESSAGE_TABLE_IDS.length; id_num++){
+      var table_name = CUSTOM_MESSAGE_TABLE_IDS[id_num];
+      for(var i = 0; i < used_languages.length; i++){
+        var language_value = $(table_name + '_language_value_' + (i+1)).value;
+        var used_language = used_languages[i];
+        languages_to_values[used_language] = language_value;
+      }
+      $(table_name).value = JSON.stringify(languages_to_values);
+    }
     return true;
   }
 
+  function change_text(row_number) {
+    var language_codes = $('repo_titles_language_selection_' + row_number);
+    var selected_index = language_codes.selectedIndex;
+    var language = language_codes[selected_index].text;
+    var language_selected_elements = document.getElementsByClassName(
+                                     'language_selected_' + row_number);
+    for (var i = 0; i < language_selected_elements.length; i++) {
+      language_selected_elements[i].innerHTML = language;
+    }
+  }
+
   /* Adds a row to the languages table so that the user can add a new language
    * and give it a title.
    * Returns the number of the row that was added
    */
   function add_language() {
-    var table = $('language_table');
+    var table_id = 'repo_titles';
```

Comment:
Let's not define table_id. We should just use constants like 'repo_titles_table' and 'repo_titles_language_selection_'.

Why? That way, one can search within this file for keyword like 'repo_titles_language_selection_' and understand how code is structured. In the current way it's written, it's very difficult to do so.